### DISCUSSION TITLE
gRPC disruptions in non-Golang apps

### DISCUSSION THREAD

**Discussion by nikos912000** [DataDog/chaos-controller#516] [Ideas, user requests and proposals]
I was curious if it easy to execute gRPC Disruptions vs apps not written in Go.

Do you have any examples that you could share?

From a quick look in the implementation:
- One would need the proto file + the codegen. This is straightforward to port to different languages.
- Also, the logic from the [chaos_service](https://github.com/DataDog/chaos-controller/blob/main/grpc/chaos_service.go); this includes the implementation of the interfaces plus the interceptor. For e.g. Java I suspect the interceptor will be an implementation of `io.grpc.ServerInterceptor` like [this](https://github.com/nikos912000/grpcdemo/blob/master/service/src/main/java/com/nkatirtzis/grpcdemo/interceptors/GrpcExceptionInterceptor.java). This part would require some work.
- Additional logic that gets used in the listener should also be ported to the target language. E.g. the [calculations](https://github.com/DataDog/chaos-controller/blob/main/grpc/calculations/calculations.go). This would also need some effort.

**Comment by ptnapoleon** [comment]
We have no use cases for this internally, and thus have never attempted it, unfortunately. I think it is possible, as you note, because all of the server-side code could be rewritten in the target language, however I definitely don't have any examples to share.

**Reply by nikos912000** [reply]
No worries. We can give this a try and if it works we'll share examples :)
